tests/pyfunc/sample_code/python_model_decorator.py

```diff
@@ -0,0 +1,11 @@
+from mlflow.models import set_model
+from mlflow.pyfunc import PythonModel
+
+
+@set_model
+class MyModel(PythonModel):
+    def predict(self, context, model_input):
+        return f"This was the input: {model_input}"
+
+
+MyModel()
```

Comment:
How is this different from the following code?

```python
class MyModel(PythonModel):
    def predict(self, context, model_input):
        return f"This was the input: {model_input}"


set_model(MyModel())
```

What's the benefit of using `set_model` as a decorator?

Reply:
+1, do customers prefer decorators over direct API call?

Reply:
The motivation is that customers are not super familiar with this paradigm of setting a model this way by calling a function.
users are more familiar with decorators now, example for creating a step in langgraph, or annotating a trace. So was wondering if we can create a similar API for the same.

What do you think?
I am down to create a new name like `@model` or something but that module is already exposed in MLflow so can't reuse that :( 
Any name suggestions?